File: mapir-android-service-sdk/src/main/java/ir/map/servicesdk/response/EstimatedTimeArrivalResponse.java
package ir.map.servicesdk.response;

import org.json.JSONArray;
import org.json.JSONObject;

import java.util.ArrayList;
import java.util.List;

import ir.map.servicesdk.model.base.MapirResponse;
import ir.map.servicesdk.model.inner.ETALeg;

public class EstimatedTimeArrivalResponse extends MapirResponse {

    private Double distance;
    private Double duration;
    private List<ETALeg> legs;

    private EstimatedTimeArrivalResponse(Double distance, Double duration, List<ETALeg> legs) {
        this.distance = distance;
        this.duration = duration;
        this.legs = legs;
    }

    public static MapirResponse createEstimatedTimeArrivalResponse(String data) {
        try {
            JSONObject tempData = new JSONObject(data);
            JSONObject tempRoutes = new JSONObject(tempData.get("routes").toString());

            JSONArray tempLegs = new JSONArray(tempRoutes.get("legs").toString());
            List<ETALeg> legItems = new ArrayList<>();
            for (int i = 0; i < tempLegs.length(); i++) {
                JSONObject tempLeg = new JSONObject(tempLegs.get(i).toString());

                legItems.add(
                        new ETALeg(
                                tempLeg.getDouble("distance"),
                                tempLeg.getDouble("duration")
                        )
                );
            }

            return new EstimatedTimeArrivalResponse(
                    tempRoutes.getDouble("distance"),
                    tempRoutes.getDouble("duration"),
                    legItems
            );
        } catch (Exception e) {
            return null;
        }
    }

    public Double getDistance() {
        return distance;
    }

    public Double getDuration() {
        return duration;
    }

    public List<ETALeg> getLegs() {
        return legs;
    }
}

File: mapir-android-service-sdk-ktx/src/main/java/ir/map/servicesdk/response/EstimatedTimeArrivalResponse.kt
package ir.map.servicesdk.response

import ir.map.servicesdk.model.base.MapirResponse
import ir.map.servicesdk.model.inner.ETALeg
import org.json.JSONArray
import org.json.JSONObject
import java.util.*

class EstimatedTimeArrivalResponse private constructor(val distance: Double, val duration: Double, val legs: List<ETALeg>) : MapirResponse() {
    companion object {
        fun createEstimatedTimeArrivalResponse(data: String?) = try {
            val tempData = JSONObject(data)
            val tempRoutes = JSONObject(tempData["routes"].toString())
            val tempLegs = JSONArray(tempRoutes["legs"].toString())
            val legItems: MutableList<ETALeg> = ArrayList<ETALeg>()
            for (i in 0 until tempLegs.length()) {
                val tempLeg = JSONObject(tempLegs[i].toString())
                legItems.add(
                        ETALeg(
                                tempLeg.getDouble("distance"),
                                tempLeg.getDouble("duration")
                        )
                )
            }
            EstimatedTimeArrivalResponse(
                    tempRoutes.getDouble("distance"),
                    tempRoutes.getDouble("duration"),
                    legItems
            )
        } catch (e: Exception) {
            null
        }
    }
}